commit d055481ce39d03652ac60de5078889e15b6917ff
Author: Florian Weimer <fweimer@redhat.com>
Date:   Mon May 16 21:59:24 2022 +0200

    scripts/glibcelf.py: Add *T_RISCV_* constants
    
    SHT_RISCV_ATTRIBUTES, PT_RISCV_ATTRIBUTES, DT_RISCV_VARIANT_CC were
    added in commit 0b6c6750732483b4d59c2fcb45484079cd84157d
    ("Update RISC-V specific ELF definitions").  This caused the
    elf/tst-glibcelf consistency check to fail.
    
    Reviewed-by: Adhemerval Zanella  <adhemerval.zanella@linaro.org>

diff --git a/scripts/glibcelf.py b/scripts/glibcelf.py
index f847b36c55c15b8a..07bef940433b4c99 100644
--- a/scripts/glibcelf.py
+++ b/scripts/glibcelf.py
@@ -385,6 +385,10 @@ class ShtPARISC(enum.Enum):
     SHT_PARISC_UNWIND = 0x70000001
     SHT_PARISC_DOC = 0x70000002
 
+class ShtRISCV(enum.Enum):
+    """Supplemental SHT_* constants for EM_RISCV."""
+    SHT_RISCV_ATTRIBUTES = 0x70000003
+
 class Pf(enum.IntFlag):
     """Program header flags.  Type of Phdr.p_flags values."""
     PF_X = 1
@@ -558,6 +562,10 @@ class PtPARISC(enum.Enum):
     PT_PARISC_ARCHEXT = 0x70000000
     PT_PARISC_UNWIND = 0x70000001
 
+class PtRISCV(enum.Enum):
+    """Supplemental PT_* constants for EM_RISCV."""
+    PT_RISCV_ATTRIBUTES = 0x70000003
+
 class Dt(_OpenIntEnum):
     """ELF dynamic segment tags.  Type of Dyn.d_val."""
     DT_NULL = 0
@@ -710,6 +718,10 @@ class DtPPC64(enum.Enum):
     DT_PPC64_OPDSZ = 0x70000002
     DT_PPC64_OPT = 0x70000003
 
+class DtRISCV(enum.Enum):
+    """Supplemental DT_* constants for EM_RISCV."""
+    DT_RISCV_VARIANT_CC = 0x70000001
+
 class DtSPARC(enum.Enum):
     """Supplemental DT_* constants for EM_SPARC."""
     DT_SPARC_REGISTER = 0x70000001
